libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2252,3 +2252,102 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 {
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
+
+string YulUtilFunctions::tryDecodeErrorMessageFunction()
+{
+	string const functionName = "try_decode_error_message";
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		string const errorHash = FixedHash<4>(util::keccak256("Error(string)")).hex();
+		return util::Whiskers(R"(
+			function <functionName>() -> data {
+				data := mload(<freeMemoryPointer>)
+
+				if lt(returndatasize(), 0x44) {
+					data := 0
```

Comment:
Then these can be removed.

Reply:
Maybe add a new local variable and only assign to return variable in the success case.